setup.py

```diff
@@ -1,17 +1,26 @@
+import sys
+
 from distutils.core import setup
 
-import sexpdata
+with open('README.rst') as f:
+    long_description = f.read()
+
+install_requires = []
+if sys.version_info < (3, 4):
+    install_requires.append('singledispatch')
+if sys.version_info < (2, 7):
+    install_requires.append('ordereddict')
```

Comment:
As I mentioned before, I'm not sure how to make that work without requiring users first install a new version of setuptools.  The version/platform qualification feature shown requires a specific 20.x or newer, and I find that in a fresh virtualenv on my Ubuntu 18.10 system, setuptools 18.2 is still the installed version.  An Ubuntu LTS or RedHat user is likely to have an older version still.

To that end, I will argue for keeping something like this.  I will be updating it to use a string based compare that I think will be a bit more resilient as well as getting rid of the `ordereddict` because of dropping 2.6.

Reply:
Not sure which `version/platform` marker do you have in mind, but I thought that `python_version` should be enough. According to https://gsnedders.github.io/python-marker-test/results.html it is supported since setuptools 3.3. Or am I understanding something it incorrectly?

Reply:
A StackOverflow comment said that  wasn't usable until 20.8.1: https://stackoverflow.com/a/32643122/198480 Perhaps I misunderstood exactly what part he was referring to though.  I did check the setuptools change log to verify his comment.

When I do:
```python
install_requires = [
    'singledispatch;python_version<"3.4"',
]
```
I get this error:
```
    error in sexpdata setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Expected version spec in singledispatch;python_version<"3.4" at ;python_version<"3.4"
``` 

So, I"m not arguing with the `python-marker-test` link you provided, just saying that when I try to do what I think should be the modern way, it doesn't work, but if I force install the latest setuptools it does work.

Reply:
Ok, np, let's keep it that way.